test/app-luatest/gh_9806_http_client_version_test.lua

```diff
@@ -0,0 +1,95 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun')
+local socket = require('socket')
+
+local g = t.group()
+
+g.before_all(treegen.init)
+g.after_all(treegen.clean)
+
+g.before_each(function(g)
+    g.server = socket.tcp_server('127.0.0.1', 0, function(_s) end)
+end)
+
+g.after_each(function(g)
+    g.server:close()
+end)
+
+local http_request_script = string.dump(function()
+    local http_client = require('http.client').new()
+    local proto = os.getenv('PROTO')
+    local port = os.getenv('HTTP_SERVER_PORT')
+    local uri = proto .. '://127.0.0.1:' .. port
+    pcall(http_client.get, http_client, uri, {
```

Comment:
You didn't check `pcall` ret code. Why?

Reply:
This request will definitely fail, because our `socket.tcp_server` server just closes a connection after accepting it. I agree that it worth to comment it here.

Reply:
Added a check and a comment